src/lib/EventListener/PerformAccessCheckSubscriber.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\User\EventListener;
+
+use Ibexa\Contracts\User\Controller\AccessCheckController;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\ControllerArgumentsEvent;
+
+final class PerformAccessCheckSubscriber implements EventSubscriberInterface
+{
+    public function onControllerArgumentsEvent(ControllerArgumentsEvent $event): void
+    {
+        $controller = $event->getController();
+        if (is_array($controller) && $controller[0] instanceof AccessCheckController) {
+            $controller[0]->performAccessCheck();
+        }
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            ControllerArgumentsEvent::class => 'onControllerArgumentsEvent',
```

Comment:
Isn't [`kernel.controller`](https://symfony.com/doc/current/components/http_kernel.html#3-the-kernel-controller-event) event more appropriate?

> This event is dispatched after the controller has been resolved but before executing it. It's useful to initialize things later needed by the controller...

[`kernel.controller_arguments`](https://symfony.com/doc/current/reference/events.html#kernel-controller-arguments) has a different purpose. 

> This event is dispatched just before a controller is called. It's useful to configure the arguments that are going to be passed to the controller.